### PR TITLE
Add base scraper test:

### PR DESCRIPTION
**Pull request by arian487**
Fixes https://github.com/Recidiviz/recidiviz-data/issues/1086

### CODE REVIEW

File: recidiviz/tests/ingest/scrape/base_scraper_test.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+        self.tasks.append(task)
+
+# pylint: disable=protected-access
+
+
+class TestBaseScraper(TestCase):
+    """Tests for base_scraper"""
+
+    def setup_method(self, _):
+        ii_info = IngestInfo()
```
Comment (jamwalla) [inline recidiviz/tests/ingest/scrape/base_scraper_test.py]: dumb nit - 'ii' is ingest_info so this says ingest_info_info. also use the kwargs in the create_ methods

File: recidiviz/tests/ingest/scrape/base_scraper_test.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        self.tasks = []
+
+    def populate_data(self, content, task, ingest_info):
+        return super().populate_data(content, task, ingest_info)
+
+    #  pylint: disable=arguments-differ
+    def add_task(self, _, task: QueueRequest):
+        self.tasks.append(task)
+
+# pylint: disable=protected-access
```
Comment (jamwalla) [inline recidiviz/tests/ingest/scrape/base_scraper_test.py]: nit: spacing here

File: recidiviz/tests/ingest/scrape/base_scraper_test.py [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+        # Should send the ii_info since we chose not to persist.
+        expected_tasks = [QueueRequest(
+            scrape_type=constants.ScrapeType.BACKGROUND,
+            next_task=TEST_TASK,
+            scraper_start_time=start_time,
+            ingest_info=self.ii_info
+        )]
+
+        self.assertEqual(mock_get_more.call_count, 1)
```
Comment (jamwalla) [inline recidiviz/tests/ingest/scrape/base_scraper_test.py]: [assert_called_once](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once) / [assert_not_called](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called)?

File: recidiviz/tests/ingest/scrape/base_scraper_test.py (hunk 1 above)
Comment (jamwalla) [reply recidiviz/tests/ingest/scrape/base_scraper_test.py]: actually - any reason this couldn't be a top-level 'constant' instead of in this setup method?

Comment (arian487) [reply recidiviz/tests/ingest/scrape/base_scraper_test.py]: We call convert_ingest_info_to_proto which modifies it.  Also fixed the name

File: recidiviz/tests/ingest/scrape/base_scraper_test.py (hunk 2 above)
Comment (arian487) [reply recidiviz/tests/ingest/scrape/base_scraper_test.py]: Done.

File: recidiviz/tests/ingest/scrape/base_scraper_test.py (hunk 3 above)
Comment (arian487) [reply recidiviz/tests/ingest/scrape/base_scraper_test.py]: Did assert called once with, but pylint complains about these.  Thinks that the line has no affect...

**Comment by arian487** [comment]
The plan to test BATH_WRITES is to just test when I actually launch that (probably today)

**Comment by jamwalla** [comment]
all sounds good!

**Pull request merged**